js/privatebin.js

```diff
@@ -1975,15 +2013,11 @@ jQuery.PrivateBin = (function($, RawDeflate) {
                             return a.length - b.length;
                         })[0];
                         if (typeof shortUrl === 'string' && shortUrl.length > 0) {
-                            I18n._(
```

Comment:
I guess you mean the `$pasteUrl` element?

Yes I we only display & enable link shortener button upon paste create successfully.

https://github.com/Haocen/PrivateBin-Contribute/blob/860980421dbe4be3746bdd376156b5b61a6ea564/js/privatebin.js#L2096

As you can see in above line `PasteStatus.createPasteNotification` will unhide the `#pastesuccess` element, which contain link shortener button with a call to `$pasteSuccess.removeClass('hidden')`.

It will also save the newly created `$pasteUrl` element to the expected variable:
```
$pasteUrl = $('#pasteurl');
```

Before then user will not be able to click the button at all.

Reply:
No, I did really mean the #pastelink which surrounds the #pasteurl - I suspect #pasteurl doesn't exist if #pastelink isn't initialized. But we will know quickly if this assumption applies or not, when you set up the test site and create a paste. If it works, please accept my apologies for the distraction.